tools/SourceKit/tools/sourcekitd/lib/API/sourcekitdAPI-InProc.cpp

```diff
@@ -0,0 +1,239 @@
+//===--- sourcekitdAPI-InProc.cpp -----------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "sourcekitd/sourcekitd.h"
+#include "sourcekitd/Internal.h"
+#include "SourceKit/Support/UIdent.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/ErrorHandling.h"
+
+using namespace SourceKit;
+using namespace sourcekitd;
+using llvm::ArrayRef;
+using llvm::StringRef;
+using llvm::raw_ostream;
+
+// The following functions are declared in sourcekitd/sourcekitd.h.
+// These are their "in-process" implementations.
+
+sourcekitd_uid_t
+sourcekitd_uid_get_from_buf(const char *buf, size_t length) {
+  llvm::report_fatal_error("not yet implemented");
```

Comment:
In the process of getting things to build, I found that there were a bunch more methods that needed stub implementations beyond the ones here. A sampling of these are: `sourcekitd_uid_get_from_cstr`, `sourcekitd_request_dictionary_set_string`, `sourcekitd_request_array_set_int64`